src/core/Akka/IO/TcpSettings.cs

```diff
@@ -38,38 +38,72 @@ public static TcpSettings Create(ActorSystem system)
         /// Creates a new instance of <see cref="TcpSettings"/> class 
         /// and fills it with values parsed from provided HOCON config.
         /// </summary>
-        /// <param name="config">TBD</param>
+        /// <param name="config">The HOCON path that contains the `akka.io.tcp` section.</param>
         public static TcpSettings Create(Config config)
         {
             if (config.IsNullOrEmpty())
                 throw ConfigurationException.NullOrEmptyConfig<TcpSettings>();
 
             return new TcpSettings(
-                bufferPoolConfigPath: config.GetString("buffer-pool", "akka.io.tcp.disabled-buffer-pool"),
-                initialSocketAsyncEventArgs: config.GetInt("nr-of-socket-async-event-args", 32),
                 traceLogging: config.GetBoolean("trace-logging", false),
                 batchAcceptLimit: config.GetString("batch-accept-limit") == "scale-to-cpus"
                     ? DefaultAcceptLimit
                     : config.GetInt("batch-accept-limit", DefaultAcceptLimit),
                 registerTimeout: config.GetTimeSpan("register-timeout", TimeSpan.FromSeconds(5)),
-                receivedMessageSizeLimit: config.GetString("max-received-message-size", "unlimited") == "unlimited"
-                    ? int.MaxValue
-                    : config.GetInt("max-received-message-size", 0),
+                maxFrameSizeBytes: (int)config.GetByteSize("maximum-frame-size", 4096).Value,
+                receiveBufferSize: (int)config.GetByteSize("receive-buffer-size", 8192).Value,
+                sendBufferSize: (int)config.GetByteSize("send-buffer-size", 8192).Value,
                 managementDispatcher: config.GetString("management-dispatcher", "akka.actor.default-dispatcher"),
-                fileIoDispatcher: config.GetString("file-io-dispatcher", "akka.actor.default-dispatcher"),
-                transferToLimit: config.GetString("file-io-transferTo-limit", null) == "unlimited"
-                    ? int.MaxValue
-                    : config.GetInt("file-io-transferTo-limit", 512 * 1024),
                 finishConnectRetries: config.GetInt("finish-connect-retries", 5),
                 outgoingSocketForceIpv4: config.GetBoolean("outgoing-socket-force-ipv4", false),
                 writeCommandsQueueMaxSize: config.GetInt("write-commands-queue-max-size", -1));
         }
+        
+        
+        // private so we can change the constructor in the future
+        private TcpSettings(
```

Comment:
Breaking change: made the CTOR private. No one could possibly `new` up this class anyway, given how complicated it is. So have them use `TcpSettings.Create` plus the record syntax to manipulate the values afterwards. This will make it much easier for us to change settings values again in the future if needed.